Logs

<details>

```
error: builder for '/nix/store/5wkc1x20w39h2fgas5b714di9xxsmq92-python3.12-awkward-2.7.2.drv' failed with exit code 1;
       last 25 log lines:
       > SKIPPED [1] tests/test_3292_to_tensorflow_from_tensorflow.py:13: could not import 'tensorflow': No module named 'tensorflow'
       > SKIPPED [1] tests/test_0652_tests_of_complex_numbers.py:220: Remember to implement sorting for complex numbers.
       > SKIPPED [1] tests/test_0115_generic_reducer_operation.py:1211: I can't think of a canonical UnionArray (non-mergeable contents) that can be used in a reducer
       > SKIPPED [1] tests/test_1072_sort.py:716: I can't think of a canonical UnionArray (non-mergeable contents) that can be used in sorting
       > SKIPPED [1] tests/test_1072_sort.py:741: I can't think of a canonical UnionArray (non-mergeable contents) that can be used in sorting
       > SKIPPED [1] tests/test_1672_broadcast_parameters.py:11: string broadcasting is broken
       > SKIPPED [1] tests/test_1672_broadcast_parameters.py:25: string broadcasting is broken
       > SKIPPED [1] tests/test_1672_broadcast_parameters.py:39: string broadcasting is broken
       > SKIPPED [1] tests/test_1672_broadcast_parameters.py:57: string broadcasting is broken
       > SKIPPED [4] tests/test_1440_start_v2_to_parquet.py:224: Categorical arrays can't roundtrip through Parquet due to ARROW-14525
       > SKIPPED [1] tests/test_2297_common_backend.py:58: could not import 'ROOT': No module named 'ROOT'
       > SKIPPED [1] tests/test_1764_jax_jacobian.py:16: Jacobian support not implemented
       > SKIPPED [2] tests/test_2793_nep_50_gradual_support.py:44: NumPy >= 1.24.0 has NEP-50 support
       > SKIPPED [1] tests/test_2799_numba_ufunc_resolution.py:39: Numba has NEP-50 support
       > FAILED tests/test_1447_jax_autodiff_slices_ufuncs.py::test_regular_array_4 - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1447_jax_autodiff_slices_ufuncs.py::test_regular_array_5 - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1447_jax_autodiff_slices_ufuncs.py::test_regular_array_2 - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1447_jax_autodiff_slices_ufuncs.py::test_recordarray_4 - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1447_jax_autodiff_slices_ufuncs.py::test_recordarray_5 - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1447_jax_autodiff_slices_ufuncs.py::test_recordarray_6 - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1490_jax_reducers_combinations.py::test_reducer[min-0] - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1490_jax_reducers_combinations.py::test_reducer[min-1] - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1490_jax_reducers_combinations.py::test_reducer[max-0] - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > FAILED tests/test_1490_jax_reducers_combinations.py::test_reducer[max-1] - jax.errors.UnexpectedTracerError: Encountered an unexpected tracer. A funct...
       > ============ 10 failed, 2957 passed, 36 skipped in 63.03s (0:01:03) ============
```

</details>

--- a/pkgs/development/python-modules/awkward/default.nix
+++ b/pkgs/development/python-modules/awkward/default.nix
@@ -24,8 +24,6 @@
   pyarrow,
   pytest-xdist,
   pytestCheckHook,
-  jax,
-  jaxlib,
 
   stdenv,
 }:
@@ -61,28 +59,23 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "awkward" ];
 
-  nativeCheckInputs =
-    [
-      fsspec
-      numba
-      setuptools
-      numexpr
-      pandas
-      pyarrow
-      pytest-xdist
-      pytestCheckHook
-    ]
-    ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [
-      # no support for darwin
-      jax
-      jaxlib
-    ];
+  nativeCheckInputs = [
+    fsspec
+    numba
+    setuptools
+    numexpr
+    pandas
+    pyarrow
+    pytest-xdist
+    pytestCheckHook
+  ];
 
-  # The following tests have been disabled because they need to be run on a GPU platform.
   disabledTestPaths = [
+    # Need to be run on a GPU platform.
     "tests-cuda"
-    # Disable tests dependending on jax on darwin
-  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [ "tests/test_2603_custom_behaviors_with_jax.py" ];
+    # JAX is broken
+    "tests/test_2603_custom_behaviors_with_jax.py"
+  ];
 
   disabledTests = [
     # AssertionError: Regex pattern did not match.

--- a/pkgs/development/python-modules/correctionlib/default.nix
+++ b/pkgs/development/python-modules/correctionlib/default.nix
@@ -2,6 +2,7 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
+  fetchpatch,
 
   # build-system
   cmake,
@@ -36,6 +37,16 @@ buildPythonPackage rec {
     hash = "sha256-l+JjW/giGzU00z0jBN3D4KB/LjTIxeJb3CS+Ge0gbiA=";
     fetchSubmodules = true;
   };
+
+  patches = [
+    # fix https://github.com/Tencent/rapidjson/issues/2277
+    (fetchpatch {
+      url = "https://github.com/Tencent/rapidjson/pull/719.diff";
+      hash = "sha256-xarSfi9o73KoJo0ijT0G8fyTSYVuY0+9rLEtfUwas0Q=";
+      extraPrefix = "rapidjson/";
+      stripLen = 1;
+    })
+  ];
 
   build-system = [
     cmake

--- a/pkgs/development/python-modules/fastjet/default.nix
+++ b/pkgs/development/python-modules/fastjet/default.nix
@@ -1,0 +1,92 @@
+{
+  lib,
+  stdenv,
+  fetchPypi,
+  fetchurl,
+  buildPythonPackage,
+  pytestCheckHook,
+  pkgs,
+  awkward,
+  numpy,
+  pybind11,
+  python,
+  setuptools,
+  setuptools-scm,
+  vector,
+}:
+
+let
+  fastjet =
+    (pkgs.fastjet.override {
+      inherit python;
+      withPython = true;
+    }).overrideAttrs
+      (prev: {
+        postInstall =
+          (prev.postInstall or "")
+          + ''
+            mv "$out/${python.sitePackages}/"{fastjet.py,_fastjet_swig.py}
+          '';
+      });
+  fastjet-contrib = pkgs.fastjet-contrib.override {
+    inherit fastjet;
+  };
+in
+
+buildPythonPackage rec {
+  pname = "fastjet";
+  version = "3.4.2.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    pname = "fastjet";
+    inherit version;
+    hash = "sha256-YlYJWCdwEBiG+koh1X2app1HinvktryisxP/C024g1k=";
+  };
+
+  # unvendor fastjet/fastjet-contrib
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-fail 'cmdclass={"build_ext": FastJetBuild, "install": FastJetInstall},' "" \
+      --replace-fail 'str(OUTPUT / "include")' "" \
+      --replace-fail 'str(OUTPUT / "lib")' ""
+    for file in src/fastjet/*.py; do
+      substituteInPlace "$file" \
+        --replace-warn "fastjet._swig" "_fastjet_swig"
+    done
+    sed -i src/fastjet/_pyjet.py -e '1iimport _fastjet_swig'
+  '';
+
+  strictDeps = true;
+
+  build-system = [
+    setuptools
+    setuptools-scm
+  ];
+
+  dependencies = [
+    fastjet
+  ];
+
+  buildInputs = [
+    pybind11
+    fastjet-contrib
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    awkward
+    numpy
+    vector
+  ];
+
+  env.SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  meta = {
+    description = "Jet-finding in the Scikit-HEP ecosystem";
+    homepage = "https://github.com/scikit-hep/fastjet";
+    changelog = "https://github.com/scikit-hep/fastjet/releases/tag/v${version}";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ veprbl ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4492,10 +4492,7 @@ self: super: with self; {
 
   fastimport = callPackage ../development/python-modules/fastimport { };
 
-  fastjet = toPythonModule (pkgs.fastjet.override {
-    withPython = true;
-    inherit (self) python;
-  });
+  fastjet = callPackage ../development/python-modules/fastjet { };
 
   fastjsonschema = callPackage ../development/python-modules/fastjsonschema { };
 

